File: 채현/W1/섬 연결하기.py
def solution(n, costs):
    answer = 0
    costs.sort(key = lambda x: x[2]) # 비용 오름차순
    link = set([costs[0][0]]) # 가장 적은 비용의 시작점

    while len(link) != n:
        for v in costs:
            if v[0] in link and v[1] in link:
                continue
            if v[0] in link or v[1] in link:
                link.update([v[0], v[1]])
                answer += v[2]
                break
                
    return answer

File: 주은/W1/섬_연결하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int parent[101];


void init(int N)
{
	int i;
	for (i = 0; i < N; i++) parent[i] = i;
}

int find(int x)
{
	if (parent[x] == x) return x;
	return parent[x] = find(parent[x]);
}

void uni(int a, int b)
{
	int ka, kb;
	ka = find(a); kb = find(b);
	parent[ka] = kb;
}

bool cmp(vector<int> a, vector<int> b) {
    return a[2] < b[2];
}
int solution(int n, vector<vector<int>> costs) {
    int answer = 0;
    init(n);
    sort(costs.begin(), costs.end(), cmp);
    
    for (int i = 0; i < costs.size(); i++) {
        if (find(costs[i][0]) != find(costs[i][1])) {
            answer += costs[i][2];
            uni(costs[i][0], costs[i][1]);
        }
    }
    
    return answer;
}